packages/vite/src/node/server/middlewares/transform.ts

```diff
@@ -147,7 +151,10 @@ export function transformMiddleware(
 
         // for CSS, we need to differentiate between normal CSS requests and
         // imports
-        if (isCSSRequest(url) && req.headers.accept?.includes('text/css')) {
+        if (
+          isIndirectCSSRequest(url) &&
```

Comment:
I think we shouldn't use `isIndirectCSSRequest` here because at this point we don't know if it is direct or not. It would be more clear to me if we use `isCSSRequest(url) && !isDirectRequest(url)`
This should read as "is this a CSS Request that is not yet marked as direct?"